include/stdexec/execution.hpp

```diff
@@ -5485,6 +5485,203 @@ namespace stdexec {
+        // make __result_ emplacement visible when __count_ goes from one to zero
+        // This relies on the fact that each sender will call notify() at most once
+        if (__count_.fetch_sub(1, std::memory_order_acq_rel) == 1) {
+          __on_stop_.reset();
+          std::visit(
+              [this]<class _Tuple>(_Tuple&& __result) {
+                if constexpr (std::same_as<std::decay_t<_Tuple>, std::monostate>) {
+                  set_stopped((_Receiver &&) __receiver_);
+                } else {
+                  auto stop_token = get_stop_token(get_env(__receiver_));
+                  if (stop_token.stop_requested()) {
+                    set_stopped((_Receiver &&) __receiver_);
+                    return;
+                  }
+                  std::apply(
+                      [this]<class _C, class... _As>(_C, _As&&... __args) noexcept {
+                        _C{}((_Receiver &&) __receiver_, (_As &&) __args...);
+                      },
+                      (_Tuple &&) __result);
+                }
+              },
+              (__result_type &&) __result_);
+        }
+      }
+
+      friend void tag_invoke(start_t, __t& __self) noexcept {
+        __self.__on_stop_.emplace(get_stop_token(get_env(__self.__receiver_)),
+                                  __on_stop_requested{__self.__stop_source_});
+        if (__self.__stop_source_.stop_requested()) {
+          stdexec::set_stopped((_Receiver &&) __self.__receiver_);
+        } else {
+          std::apply([](auto&... __ops) { (start(__ops), ...); }, __self.__ops_);
+        }
+      }
+    };
+  };
+  template <class _Receiver, class... _Senders>
+  using __op = __t<__op_id<_Receiver, _Senders...>>;
+
+  template <class _Sender, class... _Senders> struct __sender_id {
+    struct __t {
+      using __id = __sender_id;
+
+      std::tuple<_Sender, _Senders...> __senders_;
+
+      template <class _R>
+      friend __op<std::remove_cvref_t<_R>, _Sender, _Senders...> 
+        tag_invoke(connect_t, __t&& __self, _R&& __rcvr) noexcept {
+          return {(std::tuple<_Sender, _Senders...> &&) __self.__senders_, (_R &&) __rcvr};
+        }
+
+      friend __empty_env tag_invoke(get_env_t, const __t& __self) noexcept {
+        return {};
+      }
+
+      template <__decays_to<__t> _Self, class _Env>
+      friend auto tag_invoke(get_completion_signatures_t, _Self&& __self, _Env __env) noexcept
+         -> dependent_completion_signatures<_Env>;
+
+      template <__decays_to<__t> _Self, class _Env>
+      friend auto tag_invoke(get_completion_signatures_t, _Self&& __self, _Env __env) noexcept
+         -> __completion_signatures_t<_Env, _Sender, _Senders...> requires true;
+    };
+  };
+
+  template <class... _Senders> using __sender = __t<__sender_id<_Senders...>>;
+
+  struct __when_any_t {
+    template <class... _Senders>
+      requires ((stdexec::sender<_Senders> && ...) && sizeof...(_Senders) > 0)
+    auto operator()(_Senders&&... __senders) const
```

Comment:
Prefer explicitly specifying return types. If you do, then you can `decltype` the function call expression without instantiating the function template. That's a compile-time win, and it can also prevent hard errors when the function is under-constrained.

Reply:
Good to know!